src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -0,0 +1,414 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        
+        private class HillClimbing
+        {
+            private struct Complex
```

Comment:
nit: I usually prefer to put small helper class at the bottom of the scope (or moderately large ones in a separate file) so that it doesn't get in the way of the main functionality of this file, as these small things can sometimes add up to pages of code that is usually not of interest